src/Events/Editor/Full_Site/Controller.php

```diff
@@ -0,0 +1,399 @@
+	}
+
+
+	/**
+	 * Removes registered actions.
+	 *
+	 * @since TBD
+	 */
+	public function remove_actions() {
+		remove_action( 'tribe_editor_register_blocks', [ $this, 'action_register_archive_template' ] );
+		remove_action( 'tribe_editor_register_blocks', [ $this, 'action_register_single_event_template' ] );
+	}
+
+	/**
+	 * Registers the Events Archive template.
+	 *
+	 * @since 5.14.2
+	 */
+	public function action_register_archive_template() {
+		return $this->container->make( Archive_Block_Template::class )->register(); // @todo move out?
+	}
+
+	/**
+	 * Registers the Single Event template.
+	 *
+	 * @since TBD
+	 */
+	public function action_register_single_event_template() {
+		return $this->container->make( Single_Block_Template::class )->register(); // @todo move out?
+	}
+
+	/**
+	 * Adds the archive template to the array of block templates.
+	 *
+	 * @since 5.14.2
+	 * @since TBD Added support for single event templates.
+	 *
+	 * @param WP_Block_Template[] $query_result Array of found block templates.
+	 * @param array               $query        {
+	 *                                          Optional. Arguments to retrieve templates.
+	 *
+	 * @type array                $slug__in     List of slugs to include.
+	 * @type int                  $wp_id        Post ID of customized template.
+	 * }
+	 *
+	 *
+	 * @return array The modified $query.
+	 */
+	public function filter_include_templates( $query_result, $query, $template_type ) {
+		if ( ! is_array( $query_result ) ) {
+			return $query_result;
+		}
+		// Get our block template services for this query.
+		$template_services = $this->get_filtered_block_templates( $template_type );
+		foreach ( $template_services as $template ) {
+			if (
+				empty( $query['slug__in'] )
+				|| in_array( $template->slug(), $query['slug__in'], true )
+			) {
+				/**
+				 * @var WP_Block_Template $wp_template
+				 */
+				$wp_template = $template->get_block_template();
+				if ( $wp_template ) {
+					$query_result[] = $wp_template;
+				}
+			}
+		}
+
+		return $query_result;
+	}
+
+	/**
+	 * Fetch our Block Template by ID.
+	 *
+	 * @since TBD
+	 *
+	 * @param null|WP_Block_Template $block_template The filtered template.
+	 * @param string                 $id             The block template ID.
+	 * @param string                 $template_type  The template type.
+	 *
+	 * @return null|WP_Block_Template
+	 */
+	public function filter_include_template_by_id( $block_template, $id, $template_type ) {
+		if ( ! is_null( $block_template ) ) {
+			return $block_template;
+		}
+
+		$template_services = $this->get_filtered_block_templates( $template_type );
+		foreach ( $template_services as $template ) {
+			if ( $id === $template->id() ) {
+				return $template->get_block_template();
+			}
+		}
+
+		return $block_template;
+	}
+
+	/**
+	 * @param string $template_type The type of templates we are fetching.
```

Comment:
Let's add a fully qualified docblock.

Reply:
Did resolve, but something seems off with this CR - it has been updated but still shows unchanged. https://github.com/the-events-calendar/the-events-calendar/pull/4387/commits/80eebbe4095b418ed3f97b8a19da93bd5167e4be